### PR TITLE
Cv 241 integration redux

### PR DESCRIPTION
**Pull request by Kkr1328**
- initiate redux
- implement getCars, getCar, getCarsList, createCar, updateCar, deleteCar redux gateway
- implement getCameras, getCamera, getCamerasList, createCamera, updateCamera, deleteCamera redux gateway
- implement getDrivers, getDriver, getDriversList, createDriver, updateDriver, deleteDriver redux gateway
- implement getRSUs, getRSU, getRSUsList, createRSU, updateRSU, deleteRSU redux gateway

### CODE REVIEW

File: src/types/models/driver.model.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	request: IDriverRequest;
+}
+export interface IDeleteDriverQuery extends IQuerry {}
+
+export interface IDriverRequest {
+	first_name: string;
+	last_name: string;
+	phone_no: string;
+	username: string;
+	password: string;
```
Comment (ChariU) [inline src/types/models/driver.model.ts:15]: why does this interface (`IDriverRequest`) need to have `password`?

File: src/types/ENTITY.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ export interface TableRowProps {
 }
 
 export interface CameraType {
+	id: string;
```
Comment (ChariU) [inline src/types/ENTITY.tsx:16]: Is there a reason why we need both `CameraType` and `CamerasProps`, not just `CamerasProps`?

File: src/app/entity_management/drivers/page.tsx [hunk 3]
```diff
@@ -30,51 +57,169 @@ export default function Home() {
...
+		setUpdateModalData(updateData);
+		setOpenUpdateModal(true);
+	};
+	const handleCloseUpdateModal = () => {
+		setOpenUpdateModal(false);
+		setUpdateModalData(defaultData);
+	};
+	const handleUpdateNotification = () => {
+		if (!updateDriverError) {
+			enqueueSnackbar('Update a driver sucessfully', {
```
Comment (ChariU) [inline src/app/entity_management/drivers/page.tsx:118]: Just a typo error on `sucessfully`

File: src/app/entity_management/drivers/page.tsx [hunk 4]
```diff
@@ -30,51 +57,169 @@ export default function Home() {
...
+
+	// Register modal
+	const handleOpenRegisterModel = () => setOpenRegisterModal(true);
+	const handleCloseRegisterModal = () => {
+		setOpenRegisterModal(false);
+		setRegisterModalData(defaultData);
+	};
+	const handleRegisterNotification = () => {
+		if (!createDriverError) {
+			enqueueSnackbar('Register a driver sucessfully', {
```
Comment (ChariU) [inline src/app/entity_management/drivers/page.tsx:85]: typo error on `sucessfully`

File: src/app/entity_management/drivers/page.tsx [hunk 5]
```diff
@@ -30,51 +57,169 @@ export default function Home() {
...
+		setDeleteModalData(deleteData);
+		setOpenDeleteModal(true);
+	};
+	const handleCloseDeleteModal = () => {
+		setOpenDeleteModal(false);
+		setDeleteModalData(defaultData);
+	};
+	const handleDeleteNotification = () => {
+		if (!deleteDriverError) {
+			enqueueSnackbar('Delete a driver sucessfully', {
```
Comment (ChariU) [inline src/app/entity_management/drivers/page.tsx:154]: typo error on `sucessfully`

**Comment by ChariU** [comment]
There are several typo error on word `succesfully` on `src/app/entity_management/[entity]/page.tsx`, you miss another `c` before `e`.

**Review by ChariU — COMMENTED** [review]
LGTM ka! Good Job!👍

File: src/types/ENTITY.tsx (hunk 2 above)
Comment (Kkr1328) [reply src/types/ENTITY.tsx:16]: The issue of mapping between UI components and API request/response structures is still a challenge. I intend to address this problem by optimizing the types during the "Integrate" pull request, as it's a more relevant and appropriate time to tackle this issue.

File: src/app/entity_management/drivers/page.tsx (hunk 3 above)
Comment (Kkr1328) [reply src/app/entity_management/drivers/page.tsx]: resolved

File: src/app/entity_management/drivers/page.tsx (hunk 4 above)
Comment (Kkr1328) [reply src/app/entity_management/drivers/page.tsx]: resolved

File: src/app/entity_management/drivers/page.tsx (hunk 5 above)
Comment (Kkr1328) [reply src/app/entity_management/drivers/page.tsx]: resolved

File: src/types/models/driver.model.ts (hunk 1 above)
Comment (Kkr1328) [reply src/types/models/driver.model.ts:15]: IDriverRequest is used create and update user info including password

**Review by ChariU — APPROVED** [review]

**Pull request merged**